core/src/main/java/org/jruby/RubyDir.java

```diff
@@ -326,6 +326,9 @@ public static IRubyObject chdir(ThreadContext context, IRubyObject recv, IRubyOb
         if (adjustedPath.startsWith("uri:")){
             realPath = adjustedPath;
         }
+        else if (adjustedPath.startsWith("file:")){
+            realPath = "uri:" + adjustedPath;
+        }
```

Comment:
a path with ```!/``` is treated as jar uri whether this is a file: or jar: or jar:file: prefix or not. and this is the case where you want the uri: to add. but for ```uri:``` to work properly it needs ```new URL(adjustedPath).openStream()``` to work. i.e. in your case make sure there is jar:file: prefix

an path with file: onto the filesystem without the ```!/``` better just remove the file: prefix and let jruby use the filesystem.

but I do not understand why your current patch fixes the warbler problem